Write version in pyproject toml if available

mauritsvanrees opened the pull request:
- Added `pyproject.toml` and moved most metadata there. "Eat your own dog food." I am not sure if we can get rid of the remaining parts of `setup.py`.
- Add reader. I added `tomli` to the dependencies on Pythons earlier than 3.11. Chances are that you already have this via a different dependency. On 3.11 I use the built-in `tomllib`. 
- Add writer. This is not that different from how we write to `setup.py` or other files.
- Checked that `bin/prerelease` updates the version in our own `pyproject.toml`. I did not commit this.

This fixes issue #373.

reinout commented on pyproject.toml:
```diff
@@ -1,2 +1,91 @@
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
```
Add 3.11 too?

mauritsvanrees replied:
Done.

reinout commented on setup.py:
```diff
@@ -26,88 +23,9 @@ def read(filename):
-    author_email="reinout@vanrees.org",
-    url="https://zestreleaser.readthedocs.io",
-    license="GPL",
     packages=find_packages(exclude=["ez_setup"]),
```
We can probably zap `ez_setup` altogether? Wasn't that sort of a setuptools bootstrap file for when setuptools was new? I don't totally remember :-)

mauritsvanrees replied:
Right, it only makes sense to exclude `ez_setup.py` if we actually have this file, which we don't. I removed the code.

reinout commented on zest/releaser/vcs.py:
```diff
@@ -233,6 +249,50 @@ def _update_python_file_version(self, version):
         utils.write_text_file(filename, contents, encoding)
         logger.info("Set __version__ in %s to '%s'", filename, version)
 
+    def _update_pyproject_toml_version(self, version):
```
Test?

mauritsvanrees replied:
Done. See commit 1bd131523698c383d6974ded8c55a95355b498c6.

reinout reviewed (commented):
Good idea!
Most important point: I think a test has to be added? There will be corner cases later on, I fear.

mauritsvanrees commented:
Okay, so I added a test, and it works fine. The message from the commit may be interesting: 1bd131523698c383d6974ded8c55a95355b498c6.

Problem: the test does not work on Python 3.6, so I dropped support for that. We will want to make a major release for this.
And the test shows that if you *only* have a `pyproject.toml` and no `setup.py` at all, then `fullrelease` does not create and upload distributions to PyPI. So that is something that we need to fix. That is basically issue #295.

I think the current PR is fine to merge: it enables updating the version when it is in `pyproject.toml`. But releasing still needs a `setup.py` at this point, though it can be mostly empty.
We could make an 8.0.0a1 release with this.

reinout commented:
I agree. I'll merge it.
